### ISSUE TITLE
 SASS 1.63.0 cannot be imported as a module

### ISSUE THREAD

**Issue by TheLonelyAdventurer** [sass/dart-sass#1995]
SASS Version: 1.63.0
Node Version: 18.16.0

### Steps to reproduce:

Package.json:
```json
{
  "name": "minimal",
  "version": "1.0.0",
  "description": "",
  "main": "test.js",
  "type": "module",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "sass": "^1.63.0"
  }
}
```

test.js:
```js
import sass from 'sass'; 
```

### Expected
No errors.

### Log
```
(node:3040) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
(Use `node --trace-warnings ...` to show where the warning was created)
C:\***********\minimal\node_modules\sass\sass.node.js:1
import * as util from "util"
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at internalCompileFunction (node:internal/vm:73:18)
    at wrapSafe (node:internal/modules/cjs/loader:1176:20)
    at Module._compile (node:internal/modules/cjs/loader:1218:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1308:10)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:169:29)
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)

Node.js v18.16.0
```

### Notes
The same import works with 1.62.0

**Comment by nex3** [comment]
Thanks for bearing with us, folks. We changed our release infrastructure pretty significantly as part of adding support for running in the browser, and this is an unintended side effect. Expect a fix some time tomorrow afternoon PST.

**Comment by webketje** [comment]
Confirmed here too.. There are 2 issues:
The first is that package.json exports contains only `default` and `require`, yet the package.json is not `type: module`. So basically, Node will assume that it is commonjs (it is not). 

In the generated dist file `sass.dart.js` around line 53, there is a direct `require` statement that does not check whether it is actually available:
```js
if (dartNodeIsActuallyNode) {
  // This line is to:
  // 1) Prevent Webpack from bundling.
  // 2) In Webpack on Node.js, make sure we're using the native Node.js require, which is available via __non_webpack_require__
  // https://github.com/mbullington/node_preamble.dart/issues/18#issuecomment-527305561
  var url = ("undefined" !== typeof __webpack_require__ ? __non_webpack_require__ : require)("url");
```
I reran after adding the package.json `type: module` and added the following to  `sass.dart.js` and it worked:
```js
if (dartNodeIsActuallyNode) {
  // This line is to:
  // 1) Prevent Webpack from bundling.
  // 2) In Webpack on Node.js, make sure we're using the native Node.js require, which is available via __non_webpack_require__
  // https://github.com/mbullington/node_preamble.dart/issues/18#issuecomment-527305561
  var url
  if (typeof __webpack_require__ !== 'undefined') {
    url = Promise.resolve(__non_webpack_require__)('url')
  } else {
    try {
      url = import('url')
    } catch (err) {
      url = Promise.resolve(require('url'))
    }
  }

  url.then(url => { // rest of what's in the if (dartNodeIsActuallyNode)  check wrapped in this promise
```

**Comment by nex3** [comment]
> Not sure why dynamic import is not used for both commonjs & ESM modules as it has been available in Node 14+ for both and dart-sass only supports Node 14+ anyway

Dynamic import is asynchronous, and in order for Sass to remain consumable by CommonJS users it must be loadable synchronously.

**Comment by localnerve** [comment]
Was there a reason not to keep the default export for esm consumers? Seems bad to have an api divergence between to the two types of consumption.

**Comment by nex3** [comment]
We never intended to expose a default export; that's not what's documented in our TypeScript type declarations, for example. I'm surprised to hear that that's the default way of consuming a CommonJS library that just sets `exports.xyz` from ESM in Node.js. A default export is generally not something libraries expose unless it's intended to be used directly as a function or type, which it is not in Sass.

That said, I think we can add a default export as well specifically to the Node.js library. I've filed https://github.com/sass/dart-sass/issues/2008 to track that.

**Comment by ShenHongFei** [comment]
After upgrading the version, I got this error
```
import sass from 'sass';
       ^^^^
SyntaxError: The requested module 'sass' does not provide an export named 'default'
    at ModuleJob._instantiate (node:internal/modules/esm/module_job:122:21)
```

Then I changed the import method to
```ts
import * as sass from 'sass'
```

which fixed the problem

**Comment by kling90** [comment]
We had the same problem and after the update we had to change the import since it's no longer possible to import it as default in ESM.
In the documentation it says:
```js
const sass = require('sass');
``` 
Which I don't think I'm alone to simply rewrite to:
```js
import sass from 'sass';
```
Might be a good idea to add some examples for using the JS API with ESM here:
https://sass-lang.com/documentation/js-api/#md:usage

**Comment by noraj** [comment]
> We had the same problem and after the update we had to change the import since it's no longer possible to import it as default in ESM. In the documentation it says:
> 
> ```js
> const sass = require('sass');
> ```
> 
> Which I don't think I'm alone to simply rewrite to:
> 
> ```js
> import sass from 'sass';
> ```
> 
> Might be a good idea to add some examples for using the JS API with ESM here: https://sass-lang.com/documentation/js-api/#md:usage

Yeah is [my gist](https://gist.github.com/noraj/007a943dc781dc8dd3198a29205bae04) moving from CJS to EJS for gulpfiles.

**Comment by localnerve** [comment]
@nex3 
Sorry I think I've inadvertently misled here. 

It's perhaps not just the default export that is the issue, rather the **api** is different for node applications *depending on how the library is loaded* (esm vs cjs).

This is my current 1.63.3 experience

**CJS**

```javascript
const sass = require('sass')
// this works (reading sass.types)
const sassNull = sass.types.Null
```
**ESM**

```javascript
import * as sass from 'sass'
// this doesn't work (reading sass.types)
const sassNull = sass.types.Null
  // TypeError: Cannot read properties of undefined (reading 'Null')
```

So maybe coincidentally this is a default export issue, but my experience is that the api has changed for node applications *depending on how the library is loaded*

This becomes a non-trivial divergence issue for downstream consumers that provide sass helper libraries that use dart-sass that support CJS and ESM loading.

**Comment by localnerve** [comment]
> @nex3 Sorry I think I've inadvertently misled here.
> 
> It's perhaps not just the default export that is the issue, rather the **api** is different for node applications _depending on how the library is loaded_ (esm vs cjs).
> 
> This is my current 1.63.3 experience
> 
> **CJS**
> 
> ```js
> const sass = require('sass')
> // this works (reading sass.types)
> const sassNull = sass.types.Null
> ```
> 
> **ESM**
> 
> ```js
> import * as sass from 'sass'
> // this doesn't work (reading sass.types)
> const sassNull = sass.types.Null
>   // TypeError: Cannot read properties of undefined (reading 'Null')
> ```
> 
> So maybe coincidentally this is a default export issue, but my experience is that the api has changed for node applications _depending on how the library is loaded_
> 
> This becomes a non-trivial divergence issue for downstream consumers that provide sass helper libraries that use dart-sass that support CJS and ESM loading.

This appears to be fully addressed in #2010
